src/composables/accounts.ts

```diff
@@ -87,14 +92,22 @@ export function useAccounts({ store }: IDefaultComposableOptions) {
     return accounts.value.find((acc) => acc.address === address);
   }
 
+  function getAccountByGlobalIdx(index: number): IAccount | undefined {
+    return accounts.value.find((acc) => acc.globalIdx === index);
+  }
+
   function setActiveAccountByIdx(idx: number = 0) {
     // TODO replace with updating local state after removing the Vuex
-    store.commit('accounts/setActiveIdx', +(accounts.value[idx].globalIndex || 0));
+    store.commit('accounts/setActiveIdx', +(accounts.value[idx].globalIdx || 0));
   }
 
   function setActiveAccountByAddress(address?: Encoded.AccountAddress) {
     if (address) {
-      setActiveAccountByIdx(getAccountByAddress(address)?.globalIndex);
+      const account = getAccountByAddress(address);
+      if (account) {
+        setActiveAccountByIdx(account.globalIdx);
+        protocolLastUsedGlobalIdx[account.protocol] = account.globalIdx;
```

Comment:
I believe this should be called inside `setActiveAccountByIdx`
 Since we can change active accounts by calling it directly

Reply:
Martin is right. Now the logic is broken. I did test the PR, the last active account doesn't get updated.

Reply:
You have right. I changed the logic. Please retest @subhod-i